var/spack/repos/builtin/packages/intel-parallel-studio/package.py

```diff
@@ -142,3 +154,122 @@ def install(self, spec, prefix):
+                             join_path(self.prefix, 'lib', 'mic'))
+        run_env.prepend_path('MIC_LIBRARY_PATH',
+                             join_path(self.prefix, 'tbb', 'lib', 'mic'))
+        run_env.prepend_path('MIC_LD_LIBRARY_PATH',
+                             join_path(self.prefix, 'tbb', 'lib', 'mic'))
+
+        if self.spec.satisfies('+all'):
+            run_env.prepend_path('PATH',
+                                 join_path(self.prefix,
+                                           'debugger_{0}'.format(major_ver),
+                                           'gdb', 'intel64_mic', 'bin'))
+            run_env.prepend_path('LD_LIBRARY_PATH',
+                                 join_path(self.prefix,
+                                           'debugger_{0}'.format(major_ver),
+                                           'libipt', 'intel64', 'lib'))
+            run_env.set('GDBSERVER_MIC',
+                        join_path(self.prefix,
+                                  'debugger_{0}'.format(major_ver), 'gdb',
+                                  'targets', 'mic', 'bin', 'gdbserver'))
+            run_env.set('GDB_CROSS',
+                        join_path(self.prefix,
+                                  'debugger_{0}'.format(major_ver),
+                                  'gdb', 'intel64_mic', 'bin', 'gdb-mic'))
+            run_env.set('MPM_LAUNCHER',
+                        join_path(self.prefix,
+                                  'debugger_{0}'.format(major_ver), 'mpm',
+                                  'mic',
+                                  'bin', 'start_mpm.sh'))
+            run_env.set('INTEL_PYTHONHOME',
+                        join_path(self.prefix,
+                                  'debugger_{0}'.format(major_ver), 'python',
+                                  'intel64'))
+
+        if (self.spec.satisfies('+all') or self.spec.satisfies('+mpi')) and \
+           self.spec.satisfies('@cluster'):
+            run_env.prepend_path('PATH',
+                                 join_path(self.prefix, 'mpi', 'intel64',
+                                           'bin'))
+            run_env.prepend_path('LD_LIBRARY_PATH',
+                                 join_path(self.prefix, 'mpi', 'intel64',
+                                           'lib'))
+            run_env.prepend_path('LIBRARY_PATH',
+                                 join_path(self.prefix, 'mpi', 'intel64',
+                                           'lib'))
+            run_env.prepend_path('LD_LIBRARY_PATH',
+                                 join_path(self.prefix, 'mpi', 'mic', 'lib'))
+            run_env.prepend_path('MIC_LIBRARY_PATH',
+                                 join_path(self.prefix, 'mpi', 'mic', 'lib'))
+            run_env.prepend_path('MIC_LD_LIBRARY_PATH',
+                                 join_path(self.prefix, 'mpi', 'mic', 'lib'))
+            run_env.set('I_MPI_ROOT', join_path(self.prefix, 'mpi'))
```

Comment:
That would be a typo on my part, multiplied several times. Yes, that should be impi.

It is really not just for `@cluster` but for the composer edition I am concerned about the environment conflict with non Intel MPI installations. Basically, if someone loads the parallel studio module after loading an MPI module then the Intel MPI runtime will get picked up first if the impi environment is setup. This is less likely to be an issue with the cluster edition installed because people probably want impi. They probably compiled their code with it. In our environment, the impi runtime is less likely to be used than openmpi. One option might be to set the environment paths for the cluster edition, and professional I suppose, but only set `I_MPI_ROOT` for the composer edition. Then people could get to the impi run time via `I_MPI_ROOT`. An independent environment module could then be made for the impi run time, although I do not think that could be done from spack.